tools/Generator/Generator.cs

```diff
@@ -1934,7 +1935,7 @@ void GenerateCommands ()
 			IndentWriteLine ("internal static class NativeMethods");
 			IndentWriteLine ("{");
 			IndentLevel++;
-			IndentWriteLine ("const string VulkanLibrary = \"{0}\";\n", (platform == null || platform == "Windows") ? "vulkan-1" : "vulkan");
+			IndentWriteLine ("const string VulkanLibrary = \"{0}\";\n", (platform == null || platform == "Windows") ? "vulkan-1" : (platform == "iOS" ? "__Internal" : "vulkan"));
```

Comment:
I see we require "__Internal" here, which means we need the MoltenVK library around?

Would you mind to add a simple iOS sample app to our https://github.com/mono/VulkanSharp/tree/master/samples directory?

Reply:
You are right. We statically link MoltenVK with the iOS platform. Currently we have the problem that the Vulkan.dll does not work with MoltenVK because a Vulkan.dll.config with a dllmap to __Internal does not work. When changing the library path in code it works. Any idea how to fix this? Also, how to use different dllmap for iOS and android?

Before this problem is tackled adding a sample won't help a lot. I'll add a sample when it's working, but I'm not sure how to handle the MoltenVK framework. I guess it's not a good idea to check it in...

Reply:
`__Internal` would work in case the native library have been already loaded.

We might try to embed the MoltenVK native library to Vulkan.dll and use its name instead of `__Internal`.

For that we need to investigate the licensing. @migueldeicaza, do you know whether it is OK for us to embed native shared library of the MoltenVK project, which uses Apache license? https://github.com/KhronosGroup/MoltenVK/blob/master/LICENSE

Reply:
I think it's a great idea to embed MoltenVK. How would the dllmap look in this case? Is there a way to have different maps depending on the platform?

Reply:
Is there any news from the iOS team? Currently I have to compile different DLLs for iOS and the rest. Would be great to use the same for both...